src/main/java/teammates/ui/controller/InstructorStudentRecordsPageAction.java

```diff
@@ -116,5 +120,68 @@ private void filterFeedbackSessions(String courseId, List<FeedbackSessionAttribu
             }
         }
     }
+    
+    /**
+     * Maps emails of instructors to the comments they gave.
+     * @param comments
+     * @param instructor
+     * @param courseId
+     * @return A map with instructor email => comments mappings.
+     */
+    private Map<String, List<CommentAttributes>> mapCommentsToGiverEmail(List<CommentAttributes> comments,
+                                                             InstructorAttributes instructor) {
+        TreeMap<String, List<CommentAttributes>> giverEmailToCommentsMap =
+                new TreeMap<String, List<CommentAttributes>>();
+        // add an element representing the current instructor to allow "no comments" to display correctly
+        giverEmailToCommentsMap.put(InstructorStudentRecordsPageData.COMMENT_GIVER_NAME_THAT_COMES_FIRST,
+                                    new ArrayList<CommentAttributes>());
+        
+        for (CommentAttributes comment : comments) {
+            boolean isCurrentInstructorGiver = comment.giverEmail.equals(instructor.email);
+            String key = isCurrentInstructorGiver
+                       ? InstructorStudentRecordsPageData.COMMENT_GIVER_NAME_THAT_COMES_FIRST
+                       : comment.giverEmail;
+
+            List<CommentAttributes> commentList = giverEmailToCommentsMap.get(key);
+            if (commentList == null) {
+                commentList = new ArrayList<CommentAttributes>();
+                giverEmailToCommentsMap.put(key, commentList);
+            }
+            commentList.add(comment);
+        }
+        return giverEmailToCommentsMap;
+    }
+    
+    /**
+     * Maps emails of instructors giving the comments to their names.
+     * @param courseId
+     * @param giverEmails
+     * @return A map with instructor email => instructor name mappings.
+     */
+    private Map<String, String> mapGiverNameToGiverEmail(String courseId, Set<String> giverEmails) {
+        TreeMap<String, String> giverEmailToGiverNameMap = new TreeMap<String, String>();
```

Comment:
Yep it's possible, since we don't need the set from `keySet()` to be in sorted order (in fact `keySet()` is not used at all for this variable)